spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryProperties.java

```diff
@@ -175,100 +177,25 @@ public String toString() {
 	}
 
 	/**
-	 * Metadata properties.
+	 * @param addLabels include labels as metadata
+	 * @param labelsPrefix prefix for the labels
+	 * @param addAnnotations include annotations as metadata
+	 * @param annotationsPrefix prefix for the annotations
+	 * @param addPorts include ports as metadata
+	 * @param portsPrefix prefix for the ports, by default it is "port."
 	 */
-	public static class Metadata {
-
-		/**
-		 * When set, the Kubernetes labels of the services will be included as metadata of
-		 * the returned ServiceInstance.
-		 */
-		private boolean addLabels = true;
-
-		/**
-		 * When addLabels is set, then this will be used as a prefix to the key names in
-		 * the metadata map.
-		 */
-		private String labelsPrefix;
-
-		/**
-		 * When set, the Kubernetes annotations of the services will be included as
-		 * metadata of the returned ServiceInstance.
-		 */
-		private boolean addAnnotations = true;
-
-		/**
-		 * When addAnnotations is set, then this will be used as a prefix to the key names
-		 * in the metadata map.
-		 */
-		private String annotationsPrefix;
-
-		/**
-		 * When set, any named Kubernetes service ports will be included as metadata of
-		 * the returned ServiceInstance.
-		 */
-		private boolean addPorts = true;
-
-		/**
-		 * When addPorts is set, then this will be used as a prefix to the key names in
-		 * the metadata map.
-		 */
-		private String portsPrefix = "port.";
-
-		public boolean isAddLabels() {
-			return this.addLabels;
-		}
-
-		public void setAddLabels(boolean addLabels) {
-			this.addLabels = addLabels;
-		}
-
-		public String getLabelsPrefix() {
-			return this.labelsPrefix;
-		}
-
-		public void setLabelsPrefix(String labelsPrefix) {
-			this.labelsPrefix = labelsPrefix;
-		}
-
-		public boolean isAddAnnotations() {
-			return this.addAnnotations;
-		}
-
-		public void setAddAnnotations(boolean addAnnotations) {
-			this.addAnnotations = addAnnotations;
-		}
-
-		public String getAnnotationsPrefix() {
-			return this.annotationsPrefix;
-		}
-
-		public void setAnnotationsPrefix(String annotationsPrefix) {
-			this.annotationsPrefix = annotationsPrefix;
-		}
+	public record Metadata(@DefaultValue("true") boolean addLabels, String labelsPrefix,
+			@DefaultValue("true") boolean addAnnotations, String annotationsPrefix,
+			@DefaultValue("true") boolean addPorts, @DefaultValue("port.") String portsPrefix) {
 
-		public boolean isAddPorts() {
-			return this.addPorts;
-		}
-
-		public void setAddPorts(boolean addPorts) {
-			this.addPorts = addPorts;
-		}
-
-		public String getPortsPrefix() {
-			return this.portsPrefix;
-		}
```

Comment:
we need this one, because we need to tell Spring which constructor to use for injection, and this form of constructor in record is awesome! no need to define the arguments, and we add the annotation `@ConstructorBinding`